lib/Doctrine/Common/Cache/PhpFileCache.php

```
@@ -101,8 +93,14 @@ protected function doSave($id, $data, $lifeTime = 0)
             'data'      => $data
         );
 
-        $value  = var_export($value, true);
-        $code   = sprintf('<?php return %s;', $value);
+        $serializeValue = serialize($value);
+
+        if (strpos($serializeValue, 'r:') !== false || (is_object($data) && !method_exists($data, '__set_state'))) {
```

I'd rather throw an exception, now that I think of it.